package/MDAnalysis/lib/picklable_file_io.py

```diff
@@ -0,0 +1,244 @@
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+class TextIOPicklable(io.TextIOWrapper):
+    """Character and line based picklable file-like object.
+
+    This class provides a file-like :class:`io.TextIOWrapper` object that can
+    be pickled. Note that this only works in read mode.
+
+    Parameters
+    ----------
+    raw : FileIO object
+
+    Example
+    -------
+
+    >>> file = FileIOPicklable('filename')
+    >>> text_wrapped = TextIOPicklable(file)
+
+    See Also
+    ---------
+    FileIOPicklable
+    BufferIOPicklable
+
+
+    .. versionadded:: 2.0.0
+    """
+    def __init__(self, raw):
+        super().__init__(raw)
+        self.raw_class = raw.__class__
+
+    def __getstate__(self):
+        return self.raw_class, self.name, self.tell()
+
+    def __setstate__(self, args):
+        raw_class = args[0]
+        name = args[1]
+        # raw_class is used for further expansion this functionality to
+        # GZip files, which also requires a text wrapper.
+        raw = raw_class(name)
+        super().__init__(raw)
+        self.seek(args[2])
+
+
+def pickle_open(name, mode='rt'):
+    """Open file and return a stream with pickle function implemented.
+
+    This function return either BufferIOPicklable or TextIOPicklable wrapped
+    FileIOPicklable object given different reading mode. It can be used as a
+    context manager, and replace the built-in :func:`open` function
+    in read mode that only returns an unpicklable file object.
+    In order to serialize a :class:`MDAnalysis.core.Universe`, this function can
+    used to open trajectory/topology files--a object composition approach,
+    as opposed to class inheritance, which is more flexible and easier for
+    pickle implementation for new readers.
+
+    Note
+    ----
+    Can be only used with read mode.
+
+    Parameters
+    ----------
+    name : str
+        a filename given a text or byte string.
+    mode: {'r', 'rt', 'rb'} (optional)
+        'r':  open for reading in text mode;
+        'rt': read in text mode (default);
+        'rb': read in binary mode;
+
+    Returns
+    -------
+    stream-like object: BufferIOPicklable or TextIOPicklable
+        when mode is 'r' or 'rt', returns TextIOPicklable;
+        when mode is 'rb', returns BufferIOPicklable
+
+    Raises
+    ------
+    ValueError
+        if `mode` is not one of the allowed read modes
+
+    Examples
+    -------
+    open as context manager::
+
+    >>>   with pickle_open('filename') as f:
+    >>>     line = f.readline()
+
+    open as function::
+
+    >>>   f = pickle_open('filename')
+    >>>   line = f.readline()
+    >>>   f.close()
+
+    See Also
+    --------
+    :func:`MDAnalysis.lib.util.anyopen`
+    :func:`io.open`
+
+
+    .. versionadded:: 2.0.0
+    """
+    if mode not in {'r', 'rt', 'rb'}:
```

Comment:
actually. A different idea here could be to fall back to the standard open function in case of writing.